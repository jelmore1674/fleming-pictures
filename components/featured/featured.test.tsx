import React from 'react';
import { render, screen } from '@testing-library/react';
import Featured from './featured';

const mockFilms = [
	{
		title: 'Film Title Here',
		featuredImg: '/assets/images/featured1.png',
		posterImg: '/assets/images/poster1.png',
		trailer:
			'blob:https://www.youtube.com/a5512c08-80c0-4c67-bafd-b865ac6e0f50',
		sypnosis:
			'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
	},
	{
		title: 'Another Film',
		featuredImg: '/assets/images/featured2.png',
		posterImg: '/assets/images/poster2.png',
		trailer:
			'blob:https://www.youtube.com/02c39461-4044-4c77-8f21-181bb4bec043',
		sypnosis:
			'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
	},
];

describe('featured component', () => {
	it('renders component', () => {
		const featured = render(<Featured featuredFilms={mockFilms} />);

		const featuredSection = screen.getByTestId('featured-section');
		const img = screen.getByTestId('img-container0');

		expect(featuredSection).toBeInTheDocument();
		expect(img).toBeInTheDocument();
	});
});
