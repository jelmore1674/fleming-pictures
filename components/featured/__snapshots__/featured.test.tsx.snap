// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`featured component renders component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="featured-slideshow"
        data-testid="featured-section"
      >
        <div
          class="featured-section"
          data-testid="feature-img-0"
          style="transform: translate3d(0%, 0, 0);"
        >
          <div
            class="featured-slider"
            data-testid="img-container0"
          >
            <div
              class="featured-image"
              style="background-image: url(/assets/images/featured1.png);"
            >
              <button
                class="modal-button"
                data-testid="btn-0"
                id="btn-0"
              >
                More Info
              </button>
            </div>
          </div>
          <div
            class="featured-slider"
            data-testid="img-container1"
          >
            <div
              class="featured-image"
              style="background-image: url(/assets/images/featured2.png);"
            >
              <button
                class="modal-button"
                data-testid="btn-1"
                id="btn-1"
              >
                More Info
              </button>
            </div>
          </div>
        </div>
        <div
          class="slideshowDots"
        >
          <div
            class="slideshowDot active"
            data-testid="slider-status-0"
          />
          <div
            class="slideshowDot"
            data-testid="slider-status-1"
          />
        </div>
        <div
          class="icon icon-left"
          data-testid="cursor-left"
        >
          <i
            class="fas fa-chevron-left"
          />
        </div>
        <div
          class="icon icon-right"
          data-testid="cursor-right"
        >
          <i
            class="fas fa-chevron-right"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="featured-slideshow"
      data-testid="featured-section"
    >
      <div
        class="featured-section"
        data-testid="feature-img-0"
        style="transform: translate3d(0%, 0, 0);"
      >
        <div
          class="featured-slider"
          data-testid="img-container0"
        >
          <div
            class="featured-image"
            style="background-image: url(/assets/images/featured1.png);"
          >
            <button
              class="modal-button"
              data-testid="btn-0"
              id="btn-0"
            >
              More Info
            </button>
          </div>
        </div>
        <div
          class="featured-slider"
          data-testid="img-container1"
        >
          <div
            class="featured-image"
            style="background-image: url(/assets/images/featured2.png);"
          >
            <button
              class="modal-button"
              data-testid="btn-1"
              id="btn-1"
            >
              More Info
            </button>
          </div>
        </div>
      </div>
      <div
        class="slideshowDots"
      >
        <div
          class="slideshowDot active"
          data-testid="slider-status-0"
        />
        <div
          class="slideshowDot"
          data-testid="slider-status-1"
        />
      </div>
      <div
        class="icon icon-left"
        data-testid="cursor-left"
      >
        <i
          class="fas fa-chevron-left"
        />
      </div>
      <div
        class="icon icon-right"
        data-testid="cursor-right"
      >
        <i
          class="fas fa-chevron-right"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
